nni/mutable/mutable.py

```diff
@@ -501,3 +508,608 @@ class MutableSymbol(LabeledMutable, Symbol, MutableExpression):
 
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}({self.extra_repr()})'
+
+
+class Discrete(MutableSymbol, Generic[Choice]):
+    """Choosing one from a list of values.
+
+    Parameters
+    ----------
+    values
+        The list of values to choose from.
+    distribution
+        The probability distribution of the values. Should be an array with the same length as ``values``.
+        The sum of the distribution should be 1.
```

Comment:
I think it's more convenient to use "weights".

Reply:
Good point.